fl4health/clients/basic_client.py

```diff
@@ -531,7 +532,11 @@ def setup_client(self, config: Config) -> None:
         self.num_val_samples = len(self.val_loader.dataset)  # type: ignore
 
         self.set_optimizer(config)
-        self.learning_rate = self.optimizer.defaults["lr"]
+        if "global" in self.optimizers:
```

Comment:
Should we make `learning_rate` and optional parameter and have clients only set it when it makes sense to use? That is, the only default client that needs it seems to be SCAFFOLD and we can set it by overriding `setup_client` and then doing `self.optimizers["global"].defaults["lr"]`?

Reply:
This would also avoid the "awkward" setting of the learning rate here to be the "first" optimizer in the dictionary, which I think is a bit "confusing" and sort of implicitly assumes that all optimizer learning rates are equally valid for this value?

Reply:
Yeah this is a better approach for sure